datafusion/common/src/dfschema.rs

```diff
@@ -563,29 +563,6 @@ impl DFSchema {
             .all(|(dffield, arrowfield)| dffield.name() == arrowfield.name())
     }
 
-    /// Check to see if fields in 2 Arrow schemas are compatible
-    pub fn check_arrow_schema_type_compatible(
```

Comment:
Since this is a public function, can you please put it back and mark it deprecated instead to follow the deprecation guidelines: https://datafusion.apache.org/contributor-guide/api-health.html#deprecation-guidelines